engine/src/main/java/org/terasology/world/block/family/FreeformFamily.java

```diff
@@ -16,27 +16,118 @@
 
 package org.terasology.world.block.family;
 
-import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.terasology.math.Rotation;
+import org.terasology.math.Side;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.naming.Name;
+import org.terasology.world.BlockEntityRegistry;
+import org.terasology.world.WorldProvider;
+import org.terasology.world.block.Block;
+import org.terasology.world.block.BlockBuilderHelper;
 import org.terasology.world.block.BlockUri;
+import org.terasology.world.block.loader.BlockFamilyDefinition;
+import org.terasology.world.block.shapes.BlockShape;
 
-import java.util.Collections;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Locale;
+import java.util.Map;
 
 /**
  * A freeform family is a pseudo block family that can be combined with any block shape to produce an actual block
  * family.
- *
  */
-public class FreeformFamily {
+@RegisterBlockFamily("freeform")
+@FreeFormSupported(true)
+public class FreeformFamily extends AbstractBlockFamily {
     public BlockUri uri;
-    public List<String> categories;
+    private Map<Side, Block> blocks = Maps.newEnumMap(Side.class);
+    private Block block = null;
+
+    public FreeformFamily(BlockFamilyDefinition definition, BlockShape shape, BlockBuilderHelper blockBuilder) {
+        super(definition, shape, blockBuilder);
+        BlockUri uri;
+        if (CUBE_SHAPE_URN.equals(shape.getUrn())) {
+            uri = new BlockUri(definition.getUrn());
+        } else {
+            uri = new BlockUri(definition.getUrn(), shape.getUrn());
```

Comment:
Looks like you can get rid of this conditional and maybe even the constant above if you just call this signature in every case.

Unless there's a semantic difference in the two signatures?